partitura/io/exportparangonada.py

```diff
@@ -28,40 +48,78 @@ def alignment_dicts_to_array(alignment):
     # for all dicts create an appropriate entry in an array:
     # match = 0, deletion  = 1, insertion = 2
     for no, i in enumerate(alignment):
+
         if i["label"] == "match":
             array.append((no, "0", i["score_id"], str(i["performance_id"])))
         elif i["label"] == "insertion":
             array.append((no, "2", "undefined", str(i["performance_id"])))
         elif i["label"] == "deletion":
             array.append((no, "1", i["score_id"], "undefined"))
+
     alignarray = np.array(array, dtype=fields)
 
     return alignarray
 
 
-def save_csv_for_parangonada(outdir, part, ppart, align, zalign=None, feature=None):
+@deprecated_alias(
+    spart="score_data",
+    ppart="performance_data",
+    align="alignment",
+)
+def save_parangonada_csv(
+    alignment: List[dict],
+    performance_data: Union[PerformanceLike, np.ndarray],
+    score_data: Union[ScoreLike, np.ndarray],
+    outdir: Optional[PathLike] = None,
+    zalign: Optional[List[dict]] = None,
+    feature: Optional[List[dict]] = None,
+) -> Optional[Tuple[np.ndarray]]:
     """
     Save an alignment for visualization with parangonda.
 
     Parameters
     ----------
-    outdir : str
+    alignment : list
+        A list of note alignment dictionaries.
+    performance_data : Performance, PerformedPart, structured ndarray
+        The performance information
+    score_data : ScoreLike
+        The musical score. A :class:`partitura.score.Score` object,
+        a :class:`partitura.score.Part`, a :class:`partitura.score.PartGroup` or
+        a list of these.
+    outdir : PathLike
         A directory to save the files into.
-    part : Part, structured ndarray
-        A score part or its note_array.
     ppart : PerformedPart, structured ndarray
         A PerformedPart or its note_array.
-    align : list
-        A list of note alignment dictionaries.
     zalign : list, optional
         A second list of note alignment dictionaries.
     feature : list, optional
         A list of expressive feature dictionaries.
 
+    Returns
+    -------
+    perf_note_array : np.ndarray
+        The performance note array. Only returned if `outdir` is None.
+    score_note_array: np.ndarray
+        The note array from the score. Only returned if `outdir` is None.
+    alignarray: np.ndarray
+    zalignarray: np.ndarray
+    featurearray: np.ndarray
     """
 
-    part = ensure_notearray(part)
-    ppart = ensure_notearray(ppart)
+    if isinstance(score_data, (Score, Iterable)):
```

Comment:
ensure_note_array handles Score and Performance objects, why is here only the first item taken? Should this be documented?

Reply:
That is true. I was worried about the renaming of the note ids by  `note_array_from_part_list`, but then I remembered that it will not do anything if there is only a single part. Updated in commit [d637b99](https://github.com/CPJKU/partitura/commit/d637b995185186565a2bf54e35b38c51bde5f641)